### PR TITLE
front: review map's style

### LINKED ISSUE #5519: Adding more labels in editor map
## Description
- Add the signal labels in the editor map (not in the reference map)
- Add the ids of the following objects in the editor map :
  - Detectors
  - Nodes
  - Buffer stops

### PR DESCRIPTION
**Pull request by sim51**
Close #5519

On all maps:
- increase spacing between two track section label
- fix the label for switches

On editor:
- display the id for detectors /  buffer stop & switch

### CODE REVIEW

File: front/src/common/Map/Layers/GeoJSONs.tsx [hunk 1]
```diff
@@ -371,6 +371,8 @@ const GeoJSONs: FC<{
   beforeId?: string;
   // When true, all layers are rendered (ie "minZoom" restrictions are ignored)
   renderAll?: boolean;
+  // Is used in the editor ?
```
Comment (SharglutDev) [inline front/src/common/Map/Layers/GeoJSONs.tsx:374]: What do you mean by that ?

**Review by SharglutDev — APPROVED** [review]
Lgtm, good job ! Tested in local. Left only one question.

Also, I find it a bit weird with the id displayed to see "detectors.xxx" when I'm in french. But it would be probably painfull to translate them.

File: front/src/common/Map/Layers/GeoJSONs.tsx (hunk 1 above)
Comment (sim51) [reply front/src/common/Map/Layers/GeoJSONs.tsx:374]: Component `GeoJSON` is used at multiple places in the app.
For this feature, I need to replace some map style only for the editor., not everywhere.
So I added this prop to know if the component is used within the editor.

For now, the only diff in style is the text label with the ID, but perhaps in futur we will have more diff.  
That's why it's name `editor` and not `addIdInLabel` or something like that.

Comment (SharglutDev) [reply front/src/common/Map/Layers/GeoJSONs.tsx:374]: Ok I was just disturbed by the "?". Maybe a proper sentence and explanation like the one you just gave me would be more understanding for a dev passing here

**Comment by sim51** [comment]
@nicolaswurtz  I checked if it was possible to push the style of the editor into a upper code layer, but it's not so easy. 

The `GeoJSON` component is used on the editor and also on the wrap map. So we can't move the code directly inside `application/editor` package due to that. 

Moreover, when we edit an entity on the editor, this entity is  hidden in the `GeoJSON`  and  we add a new layer for the selected item. To do that we call  the `getXXXXXLayerProps` functions, which should have the display than we've got on the editor.
Check https://github.com/osrd-project/osrd/blob/dev/front/src/applications/editor/tools/pointEdition/components.tsx#L446

So I need to keep the actual code ... otherwise my code for the style must be in two different place, which is worse than to have some if/else in `getXXXXXLayerProps` functions.

**Comment by nicolaswurtz** [comment]
@sim51 I understand your point of vue, especially the "best-effort" principle, but, I've to prevent extern logic to be spread in generic components, and layers are very generic components. We should discuss about it more precisely before take a decision, the issue is not mvp prioritized, so we'll wait your return ;-)

However, I tried to find myself a solution, what about only change `geojson.tsx` file with this simple way ?

```ts
function getBufferStopsLayers(context: LayerContext, prefix: string): LayerProps[] {
  const bufferStopsLayer = getBufferStopsLayerProps(context);
  return [
    {
      ...bufferStopsLayer,
      layout: {
        ...bufferStopsLayer.layout,
        'text-field': context.editor
          ? [
              'format',
              ['get', 'id'],
              { 'font-scale': 1 },
              '\n',
              {},
              ['get', 'extensions_sncf_kp'],
              { 'font-scale': 0.8 },
            ]
          : ['get', 'extensions_sncf_kp'],
      },
      id: `${prefix}geo/buffer-stop-main`,
      minzoom: POINT_ENTITIES_MIN_ZOOM,
    },
  ];
}
```

**Pull request closed without merging**